exporter/elasticsearchexporter/exporter.go

```diff
@@ -72,17 +90,62 @@ func (e *elasticsearchExporter) pushLogsData(ctx context.Context, ld pdata.Logs)
 	panic("TODO")
 }
 
+func (e *elasticsearchExporter) pushEvent(ctx context.Context, document []byte) error {
+	attempts := 1
+	body := bytes.NewReader(document)
+	item := esBulkIndexerItem{Action: createAction, Index: e.index, Body: body}
+
+	// Setup error handler. The handler handles the per item response status based on the
+	// selective ACKing in the bulk response.
+	item.OnFailure = func(ctx context.Context, item esBulkIndexerItem, resp esBulkIndexerResponseItem, err error) {
+		switch {
+		case e.maxRetries < 0 || attempts < e.maxRetries && shouldRetryEvent(resp.Status):
```

Comment:
It's not obvious to me how `maxRetries` and `Retry.Enabled` interact. Is it possible to have `Retry.Enabled` set to false while still setting `maxRetries > 1`? If so, there should probably be a check for that here, or a check in `newElasticSearchClient` that overwrites the configured `maxRetries` if `Retry.Enabled` is false. If not, a comment explaining why this can't arise would be nice :-)

Reply:
if `Retry.Enabled` is false, then events should be dropped immediately on error. I used to have a check somewhere. Let me check and maybe add more tests if necessary to verify `Retry.Enabled=false` really disables all retry attempts.

Reply:
renaming `maxRetries` to `maxAttempts`. The `maxAttempts` is actually configured in `newExporter`. In case retry is disabled we set `maxAttempts = 1`. On Failure the predicate always evaluates to false thanks to the proper initialization of `maxAttempts`.

Note: the newElasticsearchClient configures the HTTP client. If the http request did succeed, we will have the bulk indexer report the per item status to `OnSuccess` and `OnFailure`. The client handles the HTTP retries for us, but the per item retries we have to handle ourselves.